File: android/entities/src/main/java/com/tiagohs/entities/timeline/TimelineTitle.kt
package com.tiagohs.entities.timeline

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class TimelineTitle(

    @SerializedName("id")
    val id: Int,

    @SerializedName("title")
    val title: String,

    @SerializedName("page_title")
    val pageTitle: String,

    @SerializedName("next")
    val next: String? = null,

    @SerializedName("coming_soon")
    val comingSoon: Boolean? = null,

    @SerializedName("previous")
    val previous: String? = null
): Timeline(), Serializable

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Timelines/TimelineTitle.swift
//
//  TimelineTitle.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

class TimelineTitle: Timeline {
    var id: Int!
    var title: String!
    var pageTitle: String!
    var next: String?
    var comingSoon: Bool?
    var previous: String?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        super.mapping(map: map)
        
        id                                      <- map["id"]
        title                                   <- map["title"]
        pageTitle                               <- map["page_title"]
        next                                    <- map["next"]
        comingSoon                              <- map["coming_soon"]
        previous                                <- map["previous"]
    }
}
